### PR TITLE
Add custom themes

### PR DESCRIPTION
**Pull request by RisingGlitch**
Adds option in settings to:

Create theme,
Import theme,
Delete theme,
and Export the current theme files.

Create theme adds a theme based on what you name it to the theme options list in the Themes! tab. The theme doesn't have anything until you use the import theme option.

Import theme takes a css file from you computer and assigns the css to the current theme.

Delete theme removes the theme and the css.

Export theme exports the css file.

Also implements fixes by Incognito from [here](https://github.com/UseInterstellar/Interstellar/pull/755)

### CODE REVIEW

File: static/assets/scripts/s.js [hunk 1]
```diff
@@ -240,11 +240,123 @@ switches.addEventListener("change", (event) => {
...
+
+loadCustomThemes()
+
+let themeId = localStorage.getItem("theme")
 if (themeId == "") {
   themeId = "d"
 }
 
+if (themeId == "catppuccinMocha" || themeId == "catppuccinMacchiato" || themeId == "catppuccinFrappe" || themeId == "catppuccinLatte" || themeId == "d") {
+  document.getElementById("currentThemeText").innerHTML = "Selected Theme: Default Themes"
```
Comment (IncognitoTGT) [inline static/assets/scripts/s.js:345]: do `textContent` as `innerHTML` could let you do xss

**Review by IncognitoTGT — CHANGES_REQUESTED** [review]
changed

File: static/assets/scripts/s.js (hunk 1 above)
Comment (RisingGlitch) [reply static/assets/scripts/s.js:345]: Will change, but the user sets this value themselves, so it wouldn't be a security risk

**Review by IncognitoTGT — APPROVED** [review]
lgtm

**Pull request merged**